src/ClientData/include/ClientData/ScopeIdProvider.h

```diff
@@ -49,6 +53,11 @@ class NameEqualityScopeIdProvider : public ScopeIdProvider {
   [[nodiscard]] std::optional<ScopeId> FunctionIdToScopeId(uint64_t function_id) const override;
   [[nodiscard]] uint64_t ScopeIdToFunctionId(ScopeId scope_id) const override;
 
+  [[nodiscard]] ScopeId GetMaxId() const override {
+    absl::ReaderMutexLock reader_lock{&mutex_};
+    return ScopeId(next_id_ - 1);
```

Comment:
What happens if next_id_ is 0? Would this break your logic?

Reply:
`next_id_` is at least 1, as 0 is still reserved for an invalid scope id. So in case if we've got no dynamically instrumented functions and while no manual instrumentation had been hit, we have `next_id_ == 1` and the `MaxScopeId()` returns `ScopeId(0)`... which kind of makes sense. Anyway, it will not (should not) be called in such a case, as no scopes => nothing to add under the Live tab => nothing to insert into indices_.